src/net/fe/builderStage/ShopInventory.java

```diff
@@ -0,0 +1,118 @@
+package net.fe.builderStage;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fe.builderStage.ShopInventory;
+import net.fe.modifier.Modifier;
+import net.fe.unit.Item;
+import net.fe.unit.Unit;
+import net.fe.unit.Weapon;
+
+public class ShopInventory {
+	private static ShopInventory obj;
+
+	private Map<String, Integer> inventory;
+
+	private ShopInventory(Iterable<Modifier> mods){
+		inventory = new HashMap<String, Integer>();
+
+		for(Item i : Item.getAllItems()){
+			inventory.put(i.name, -1);
+		}
+
+		if(mods != null){
+
+			for(Modifier m : mods){
+				m.setShopLimits(inventory);
+			}
+		}
+	}
+
+	/**
+	 * Singleton constructor method
+	 * 
+	 * @param mods The modifiers used by this session.  Only loaded the first time this is called, but
+	 * there's no harm in passing it again.
+	 * 
+	 * @return
+	 */
+	public static ShopInventory GetInstance(Iterable<Modifier> mods){
+		if(obj == null){
+			obj = new ShopInventory(mods);
+		}
+
+		return obj;
+	}
+
+	/**
+	 * destroys the current instance of the inventory.  To be called when the battle starts to reset
+	 * it for the next game.
+	 */
+	public void DestroyInstance(){
```

Comment:
It's that or some form of resetting it.  The only time it's being destroyed is when the game starts in order to re-initialize it so the inventory restarts if multiple games are played on the same server.  Otherwise, it works fine.  I'd argue those two things are the same, name aside.